pam/internal/gdm/protocol.go

```diff
@@ -167,3 +168,40 @@ func (d *Data) JSON() ([]byte, error) {
 
 	return bytes, err
 }
+
+var stringifyEventDataFunc = stringifyEventDataFiltered
+
+func stringifyEventDataDebug(ed *EventData) string {
+	return ed.String()
+}
+
+func stringifyEventDataFiltered(ed *EventData) string {
+	authReq, ok := ed.GetData().(*EventData_IsAuthenticatedRequested)
+	if !ok {
+		return ed.String()
+	}
+
+	item := authReq.IsAuthenticatedRequested.GetAuthenticationData().Item
+	_, ok = item.(*authd.IARequest_AuthenticationData_Challenge)
+	if !ok {
+		return ed.String()
+	}
```

Comment:
We can use the shortened-if here :upside_down_face:

Reply:
Yep, initially I was keeping the variable around, but indeed now it's not needed. Good catch.